jobs: TestRegistryLifecycle failed

jobs.TestRegistryLifecycle [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20358164?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20358164?buildTab=artifacts#/) on master @ [63f8071d339779e63385e08467ce0bfec1d32d76](https://github.com/cockroachdb/cockroach/commits/63f8071d339779e63385e08467ce0bfec1d32d76):


```
sync.(*Cond).Wait(0x4001f14600)
	GOROOT/src/sync/cond.go:71 +0xc4
github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed.(*schedulerShard).processEvents(0x400e0fb800, {0x602a878, 0x4002ecb400})
	pkg/kv/kvserver/rangefeed/scheduler.go:476 +0x608
github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed.(*Scheduler).Start.func1({0x602a878, 0x4002ecb400})
	pkg/kv/kvserver/rangefeed/scheduler.go:249 +0xf0
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x602a878, 0x4002ecb400})
	pkg/util/stop/stopper.go:438 +0x88
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 30162
	pkg/util/stop/stopper.go:436 +0x104

goroutine 30157 [select]:
github.com/cockroachdb/cockroach/pkg/util/schedulerlatency.StartSampler.func1({0x602a878, 0x4004bdc320})
	pkg/util/schedulerlatency/sampler.go:129 +0x368
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x602a878, 0x4004bdc320})
	pkg/util/stop/stopper.go:438 +0x88
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 29476
	pkg/util/stop/stopper.go:436 +0x104

goroutine 29990 [select]:
google.golang.org/grpc/internal/transport.(*controlBuffer).get(0x4004360500, 0x1)
	external/org_golang_google_grpc/internal/transport/controlbuf.go:418 +0x150
google.golang.org/grpc/internal/transport.(*loopyWriter).run(0x4006a96af0)
	external/org_golang_google_grpc/internal/transport/controlbuf.go:552 +0x70
google.golang.org/grpc/internal/transport.NewServerTransport.func2()
	external/org_golang_google_grpc/internal/transport/http2_server.go:336 +0xd8
created by google.golang.org/grpc/internal/transport.NewServerTransport in goroutine 29988
	external/org_golang_google_grpc/internal/transport/http2_server.go:333 +0x14a4

goroutine 31621 [select, 14 minutes]:
github.com/cockroachdb/cockroach/pkg/server.(*testServer).Activate.func2()
	pkg/server/testserver.go:894 +0x84
created by github.com/cockroachdb/cockroach/pkg/server.(*testServer).Activate in goroutine 29476
	pkg/server/testserver.go:891 +0x120

goroutine 29504 [select, 14 minutes]:
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).initCancelingWorkers.func1({0x602a878, 0x4004bdc320})
	pkg/sql/distsql_running.go:250 +0x94
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x602a878, 0x4004bdc320})
	pkg/util/stop/stopper.go:438 +0x88
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 29476
	pkg/util/stop/stopper.go:436 +0x104

goroutine 30138 [chan receive, 14 minutes]:
github.com/cockroachdb/cockroach/pkg/server.startListenRPCAndSQL.func4({0x4000c5fc80?, 0x602a878?})
	pkg/server/start_listen.go:167 +0x80
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x602a878, 0x4004bdc320})
	pkg/util/stop/stopper.go:438 +0x88
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 29476
	pkg/util/stop/stopper.go:436 +0x104
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/jobs @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegistryLifecycle.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-53595

jobs.TestRegistryLifecycle [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20364360?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20364360?buildTab=artifacts#/) on master @ [a73404be238e023445053272e7374167791a0886](https://github.com/cockroachdb/cockroach/commits/a73404be238e023445053272e7374167791a0886):


```
=== RUN   TestRegistryLifecycle
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/6e7de2795b6e6fc10bff51d7f2bf803d/logTestRegistryLifecycle2088071044
    test_log_scope.go:82: use -show-logs to present logs inline
--- FAIL: TestRegistryLifecycle (1.69s)
panic: runtime error: slice bounds out of range [-1:] [recovered]
	panic: runtime error: slice bounds out of range [-1:]

goroutine 7534 [running]:
testing.tRunner.func1.2({0x5012f80, 0x400f773f50})
	GOROOT/src/testing/testing.go:1734 +0x1ac
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1737 +0x334
panic({0x5012f80?, 0x400f773f50?})
	GOROOT/src/runtime/panic.go:792 +0x124
github.com/cockroachdb/cockroach/pkg/util/log/logtestutils.(*StructuredLogSpy[...]).GetUnreadLogs(0x60c57a0, 0x1)
	pkg/util/log/logtestutils/structured_log_spy.go:259 +0x1f4
github.com/cockroachdb/cockroach/pkg/jobs_test.TestRegistryLifecycle.func3.(*registryTestSuite).checkStateChangeLog.2()
	pkg/jobs/jobs_test.go:374 +0x6c
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1()
	pkg/testutils/soon.go:71 +0x40
github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration(0xa7a358200, 0x400227fcd0)
	pkg/util/retry/retry.go:346 +0xbc
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError(0x400227fef0, 0xa7a358200)
	pkg/testutils/soon.go:77 +0x54
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x6009dc8, 0x400b1a9a40}, 0x400227fef0, 0xa7a358200)
	pkg/testutils/soon.go:56 +0x44
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x6009dc8, 0x400b1a9a40}, 0x400227fef0)
	pkg/testutils/soon.go:38 +0x4c
github.com/cockroachdb/cockroach/pkg/jobs_test.(*registryTestSuite).checkStateChangeLog(...)
	pkg/jobs/jobs_test.go:373
github.com/cockroachdb/cockroach/pkg/jobs_test.TestRegistryLifecycle.func3(0x400b1a9a40)
	pkg/jobs/jobs_test.go:460 +0x31c
testing.tRunner(0x400b1a9a40, 0x574b3c8)
	GOROOT/src/testing/testing.go:1792 +0xe4
created by testing.(*T).Run in goroutine 2949
	GOROOT/src/testing/testing.go:1851 +0x374
=== RUN   TestRegistryLifecycle/pause_running
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    jobs_test.go:256: Starting resume
    jobs_test.go:275: Exiting resume
--- FAIL: TestRegistryLifecycle/pause_running (0.66s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegistryLifecycle.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

jobs.TestRegistryLifecycle [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20371358?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20371358?buildTab=artifacts#/) on master @ [064aa707cdc0283bea483b513e6d6df97219ca93](https://github.com/cockroachdb/cockroach/commits/064aa707cdc0283bea483b513e6d6df97219ca93):


```
=== RUN   TestRegistryLifecycle
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/6e7de2795b6e6fc10bff51d7f2bf803d/logTestRegistryLifecycle3123985642
    test_log_scope.go:82: use -show-logs to present logs inline
--- FAIL: TestRegistryLifecycle (2.63s)
panic: runtime error: slice bounds out of range [-1:] [recovered]
	panic: runtime error: slice bounds out of range [-1:]

goroutine 7972 [running]:
testing.tRunner.func1.2({0x51930c0, 0x400a0d0108})
	GOROOT/src/testing/testing.go:1632 +0x1bc
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1635 +0x334
panic({0x51930c0?, 0x400a0d0108?})
	GOROOT/src/runtime/panic.go:791 +0x124
github.com/cockroachdb/cockroach/pkg/util/log/logtestutils.(*StructuredLogSpy[...]).GetUnreadLogs(0x621d960, 0x1)
	pkg/util/log/logtestutils/structured_log_spy.go:259 +0x20c
github.com/cockroachdb/cockroach/pkg/jobs_test.TestRegistryLifecycle.func3.(*registryTestSuite).checkStateChangeLog.2()
	pkg/jobs/jobs_test.go:374 +0x6c
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1()
	pkg/testutils/soon.go:71 +0x40
github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration(0xa7a358200, 0x400a70fcd0)
	pkg/util/retry/retry.go:346 +0xbc
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError(0x400a70fef0, 0xa7a358200)
	pkg/testutils/soon.go:77 +0x54
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x6161880, 0x4009d4d380}, 0x400a70fef0, 0xa7a358200)
	pkg/testutils/soon.go:56 +0x44
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x6161880, 0x4009d4d380}, 0x400a70fef0)
	pkg/testutils/soon.go:38 +0x4c
github.com/cockroachdb/cockroach/pkg/jobs_test.(*registryTestSuite).checkStateChangeLog(...)
	pkg/jobs/jobs_test.go:373
github.com/cockroachdb/cockroach/pkg/jobs_test.TestRegistryLifecycle.func3(0x4009d4d380)
	pkg/jobs/jobs_test.go:460 +0x320
testing.tRunner(0x4009d4d380, 0x58d9f20)
	GOROOT/src/testing/testing.go:1690 +0xe4
created by testing.(*T).Run in goroutine 3056
	GOROOT/src/testing/testing.go:1743 +0x314
=== RUN   TestRegistryLifecycle/pause_running
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    jobs_test.go:256: Starting resume
    jobs_test.go:275: Exiting resume
--- FAIL: TestRegistryLifecycle/pause_running (1.11s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegistryLifecycle.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

jobs.TestRegistryLifecycle [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20377254?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20377254?buildTab=artifacts#/) on master @ [ae4d42ddae04a870cb778ca5b2eaa1df95ee45fc](https://github.com/cockroachdb/cockroach/commits/ae4d42ddae04a870cb778ca5b2eaa1df95ee45fc):


```
=== RUN   TestRegistryLifecycle
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/6e7de2795b6e6fc10bff51d7f2bf803d/logTestRegistryLifecycle2587208024
    test_log_scope.go:82: use -show-logs to present logs inline
--- FAIL: TestRegistryLifecycle (8.16s)
panic: runtime error: slice bounds out of range [-1:] [recovered]
	panic: runtime error: slice bounds out of range [-1:]

goroutine 28039 [running]:
testing.tRunner.func1.2({0x51a0980, 0x4002612840})
	GOROOT/src/testing/testing.go:1632 +0x1bc
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1635 +0x334
panic({0x51a0980?, 0x4002612840?})
	GOROOT/src/runtime/panic.go:791 +0x124
github.com/cockroachdb/cockroach/pkg/util/log/logtestutils.(*StructuredLogSpy[...]).GetUnreadLogs(0x6230920, 0x1)
	pkg/util/log/logtestutils/structured_log_spy.go:259 +0x20c
github.com/cockroachdb/cockroach/pkg/jobs_test.TestRegistryLifecycle.func11.(*registryTestSuite).checkStateChangeLog.3()
	pkg/jobs/jobs_test.go:374 +0x6c
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1()
	pkg/testutils/soon.go:71 +0x40
github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration(0xa7a358200, 0x40028afb50)
	pkg/util/retry/retry.go:346 +0xbc
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError(0x40028afed0, 0xa7a358200)
	pkg/testutils/soon.go:77 +0x54
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x6174660, 0x4005678b60}, 0x40028afed0, 0xa7a358200)
	pkg/testutils/soon.go:56 +0x44
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x6174660, 0x4005678b60}, 0x40028afed0)
	pkg/testutils/soon.go:38 +0x4c
github.com/cockroachdb/cockroach/pkg/jobs_test.(*registryTestSuite).checkStateChangeLog(...)
	pkg/jobs/jobs_test.go:373
github.com/cockroachdb/cockroach/pkg/jobs_test.TestRegistryLifecycle.func11(0x4005678b60)
	pkg/jobs/jobs_test.go:852 +0x528
testing.tRunner(0x4005678b60, 0x58e8c40)
	GOROOT/src/testing/testing.go:1690 +0xe4
created by testing.(*T).Run in goroutine 3066
	GOROOT/src/testing/testing.go:1743 +0x314
=== RUN   TestRegistryLifecycle/fail_marking_success_and_fail_OnFailOrCancel
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    jobs_test.go:256: Starting resume
    jobs_test.go:275: Exiting resume
    jobs_test.go:322: Starting success
    jobs_test.go:326: Exiting success
    jobs_test.go:256: Starting resume
    jobs_test.go:275: Exiting resume
    jobs_test.go:296: Starting FailOrCancel
    jobs_test.go:316: Exiting FailOrCancel
    jobs_test.go:296: Starting FailOrCancel
    jobs_test.go:316: Exiting FailOrCancel
--- FAIL: TestRegistryLifecycle/fail_marking_success_and_fail_OnFailOrCancel (0.89s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegistryLifecycle.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

jobs.TestRegistryLifecycle [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20383850?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20383850?buildTab=artifacts#/) on master @ [ce88073f9ed0112c67818ace4ccdc74f4c4d6186](https://github.com/cockroachdb/cockroach/commits/ce88073f9ed0112c67818ace4ccdc74f4c4d6186):


```
=== RUN   TestRegistryLifecycle
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/6e7de2795b6e6fc10bff51d7f2bf803d/logTestRegistryLifecycle3489738363
    test_log_scope.go:82: use -show-logs to present logs inline
--- FAIL: TestRegistryLifecycle (5.56s)
panic: runtime error: slice bounds out of range [-1:] [recovered]
	panic: runtime error: slice bounds out of range [-1:]

goroutine 27338 [running]:
testing.tRunner.func1.2({0x51a1fe0, 0x4001056918})
	GOROOT/src/testing/testing.go:1632 +0x1bc
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1635 +0x334
panic({0x51a1fe0?, 0x4001056918?})
	GOROOT/src/runtime/panic.go:791 +0x124
github.com/cockroachdb/cockroach/pkg/util/log/logtestutils.(*StructuredLogSpy[...]).GetUnreadLogs(0x6232700, 0x1)
	pkg/util/log/logtestutils/structured_log_spy.go:259 +0x20c
github.com/cockroachdb/cockroach/pkg/jobs_test.TestRegistryLifecycle.func11.(*registryTestSuite).checkStateChangeLog.3()
	pkg/jobs/jobs_test.go:374 +0x6c
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1()
	pkg/testutils/soon.go:71 +0x40
github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration(0xa7a358200, 0x4004335b50)
	pkg/util/retry/retry.go:346 +0xbc
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError(0x4004335ed0, 0xa7a358200)
	pkg/testutils/soon.go:77 +0x54
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x6176400, 0x40092bc340}, 0x4004335ed0, 0xa7a358200)
	pkg/testutils/soon.go:56 +0x44
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x6176400, 0x40092bc340}, 0x4004335ed0)
	pkg/testutils/soon.go:38 +0x4c
github.com/cockroachdb/cockroach/pkg/jobs_test.(*registryTestSuite).checkStateChangeLog(...)
	pkg/jobs/jobs_test.go:373
github.com/cockroachdb/cockroach/pkg/jobs_test.TestRegistryLifecycle.func11(0x40092bc340)
	pkg/jobs/jobs_test.go:852 +0x528
testing.tRunner(0x40092bc340, 0x58ea8e0)
	GOROOT/src/testing/testing.go:1690 +0xe4
created by testing.(*T).Run in goroutine 3011
	GOROOT/src/testing/testing.go:1743 +0x314
=== RUN   TestRegistryLifecycle/fail_marking_success_and_fail_OnFailOrCancel
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    jobs_test.go:256: Starting resume
    jobs_test.go:275: Exiting resume
    jobs_test.go:322: Starting success
    jobs_test.go:326: Exiting success
    jobs_test.go:256: Starting resume
    jobs_test.go:275: Exiting resume
    jobs_test.go:296: Starting FailOrCancel
    jobs_test.go:316: Exiting FailOrCancel
    jobs_test.go:296: Starting FailOrCancel
    jobs_test.go:316: Exiting FailOrCancel
--- FAIL: TestRegistryLifecycle/fail_marking_success_and_fail_OnFailOrCancel (0.70s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegistryLifecycle.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/jobs/jobs_test.go
+++ b/pkg/jobs/jobs_test.go
@@ -869,10 +869,17 @@ func TestRegistryLifecycle(t *testing.T) {
 		defer rts.setUp(t)()
 		defer rts.tearDown()
 
+		// Pick an ID so we know which job to mess with.
+		id := rts.registry.MakeJobID()
+		rts.mockJob.JobID = id
+
 		// Inject an error in the update to move the job to "succeeded" one time.
 		var failed atomic.Value
 		failed.Store(false)
 		rts.beforeUpdate = func(orig, updated jobs.JobMetadata) error {
+			if orig.ID != id {
+				return nil
+			}
 			if updated.State == jobs.StateSucceeded && !failed.Load().(bool) {
 				failed.Store(true)
 				return errors.New("boom")

--- a/pkg/jobs/jobs_test.go
+++ b/pkg/jobs/jobs_test.go
@@ -377,7 +377,7 @@ func (rts *registryTestSuite) checkStateChangeLog(
 				jobEventsLog.PreviousStatus == string(expectedPrevState) &&
 				jobEventsLog.NewStatus == string(expectedNewState) &&
 				strings.Contains(jobEventsLog.Error, expectedError) {
-				rts.statusChangeLogSpy.SetLastNLogsAsUnread(logpb.Channel_OPS, len(logs)-i+1)
+				rts.statusChangeLogSpy.SetLastNLogsAsUnread(logpb.Channel_OPS, len(logs)-i)
 				return nil
 			}
 		}

--- a/pkg/jobs/test_helpers.go
+++ b/pkg/jobs/test_helpers.go
@@ -44,6 +44,9 @@ func TestingCreateAndStartJob(
 	c := config{
 		jobID: r.MakeJobID(),
 	}
+	if record.JobID != 0 {
+		c.jobID = record.JobID
+	}
 	for _, opt := range opts {
 		opt(&c)
 	}

